lua/nvim_lsp/tsserver.lua

```diff
@@ -0,0 +1,111 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+local fn = vim.fn
+local api = vim.api
+local need_bins = util.need_bins
+
+local default_capabilities = lsp.protocol.make_client_capabilities()
+default_capabilities.offsetEncoding = {"utf-8", "utf-16"}
+
+local bin_name = "typescript-language-server"
+
+local install_dir = util.path.join(fn.stdpath("cache"), "nvim_lsp", "typescript-language-server")
+
+local function get_install_info()
+  local bin_dir = util.path.join(install_dir, "node_modules", ".bin")
+  local bins = { bin_dir = bin_dir; install_dir = install_dir; }
+  bins.tsserver = util.path.join(bin_dir, bin_name)
+  bins.is_installed = need_bins(bins.tsserver)
+  return bins
+end
+
+local global_commands = {
+  TsServerInstall = {
+    function()
+      if get_install_info().is_installed then
+        return print('typescript-language-server is already installed')
+      end
+      skeleton.tsserver.install()
+    end;
+    description = 'Install typescript-language-server and its dependencies to stdpath("cache")/nvim_lsp/typescript-language-server';
+  };
+  TsServerInstallInfo = {
+    function()
+      local install_info = get_install_info()
+      if not install_info.is_installed then
+        return print('typescript-language-server is not installed')
+      end
+      print(vim.inspect(install_info))
+    end;
+    description = 'Print installation infor for `typescript-language-server`'
+  }
+};
+
+util.create_module_commands("typescript-language-server", global_commands)
```

Comment:
`util.create_module_commands(...,{cmd1=...}})` is neat, let's get it into core somehow.

However, nvim-lsp shouldn't be adding a bunch of random top-level commands. Maybe something like:

    :LspInstall ts

Top-level commands are for frequent actions. Having `FooInstall` and `BarInstall` cluttering up the command space makes tab-complete more noisy.

Reply:
@justinmk yeah I was going to do that after merging this in a follow up PR to generalize installation, since a pattern started to emerge.